mesonbuild/interpreter/type_checking.py

```diff
@@ -522,6 +522,12 @@ def link_whole_validator(values: T.List[T.Union[StaticLibrary, CustomTarget, Cus
 # Applies to all build_target like classes
 _ALL_TARGET_KWS: T.List[KwargInfo] = [
     OVERRIDE_OPTIONS_KW,
+    KwargInfo(
```

Comment:
Should this be pulled out so I can consume it in simd.check? Otherwise, a dict might also be useful for getting at it.

Reply:
Should we rename the other SOURCES_KW to DECLARE_DEPENDENCY_SOURCE_KW?

Reply:
Alternatively, we could teach declare_dependency about the extra types and consolidate them back into one

Reply:
I like that idea quite a bit.